sale_order_line_available_qty/models/sale_order_line.py

```diff
@@ -0,0 +1,69 @@
+# -*- encoding: utf-8 -*-
+##############################################################################
+#
+#    OpenERP, Open Source Management Solution
+#    This module copyright (C) 2015 Savoir-faire Linux
+#    (<http://www.savoirfairelinux.com>).
+#
+#    This program is free software: you can redistribute it and/or modify
+#    it under the terms of the GNU Affero General Public License as
+#    published by the Free Software Foundation, either version 3 of the
+#    License, or (at your option) any later version.
+#
+#    This program is distributed in the hope that it will be useful,
+#    but WITHOUT ANY WARRANTY; without even the implied warranty of
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#    GNU Affero General Public License for more details.
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
+
+from openerp import models, fields, api
+import openerp.addons.decimal_precision as dp
+
+
+class SaleOrderLine(models.Model):
+    _inherit = 'sale.order.line'
+
+    @api.one
+    @api.depends('product_id')
+    def compute_qty_available(self):
+        """Get the quantity available in the selected warehouse"""
+        order = self.order_id
+        self.qty_available = sum(
+            quant.qty for quant in
+            self.env['stock.quant'].search([
+                ('location_id', '=', order.warehouse_id.
+                    lot_stock_id.id),
+                ('product_id', '=', self.product_id.id),
+                '|',
+                ('reservation_id.group_id.name', 'in',
+                    [False, order.name]),
+                ('reservation_id', '=', False),
```

Comment:
This is what I did first. But the SO, you select a warehouse, so the logic here is to retrieve the available stock from that warehouse.